controllers/volumereplicationgroup_controller.go

```diff
@@ -358,22 +357,6 @@ func (v *VRGInstance) processVRG() (ctrl.Result, error) {
 		return ctrl.Result{}, nil
 	}
 
-	if err := v.validateSchedule(); err != nil {
-		v.log.Error(err, "Failed to validate the scheduling interval")
-
-		msg := "Failed to validate scheduling interval"
-		setVRGDataErrorCondition(&v.instance.Status.Conditions, v.instance.Generation, msg)
-
-		if err = v.updateVRGStatus(false); err != nil {
-			v.log.Error(err, "VRG Status update failed")
-
-			return ctrl.Result{Requeue: true}, nil
-		}
-
-		// No requeue, as there is no reconcile till user changes desired spec to a valid value
-		return ctrl.Result{}, nil
-	}
-
```

Comment:
Probably we should not remove this from VRG?

I agree with not having multiple different implementations for validating schedule in different controllers and instead better to use a single implementation probably done in utils. However, I think that it is still better to have a validation check for schedule in VRG. It is because, VRG spec has a field for schedulingInterval. Hence in this case VRG would be doing validation of its own spec field. For DRPC it would be correct to remove the validation of schedule from there as it was validating the schedule from DRPolicy spec and DRPC spec in itself does not have the schedulingInterval field.

Let me know your thoughts on this.

Reply:
vrg.spec.schedulingInterval format should still be validated.  The change to vrg's types.go file should cause it to be validated in the create/update path.  The same change was made to drpolicy's types.go file.  A test case was added for drpolicy but not duplicated for vrg.